switches/http2/HttpInterfaceHandlers.cpp

```diff
@@ -475,3 +477,86 @@ void VariableOrEventHandler::parseJsonForm(std::string content, std::vector<std:
 
 	return;
 }
+
+FileHandler::FileHandler(HttpInterface *interface): InterfaceHttpHandler(interface)
+{
+}
+
+FileHandler::~FileHandler()
+{
+}
+
+std::string FileHandler::filePath(HttpRequest *request) const
+{
+	// check that url has a single leading slash, not followed by a dot
+	// if invalid, return empty string
+	std::string url = request->getUri();
+	if (url.length() < 2 || url[0] != '/' || url[1] == '.')
+		return std::string();
+	std::string filename = url.substr(1);
+	return getInterface()->prependDocumentRoot(filename);
+}
+
+bool FileHandler::doesFileExist(HttpRequest *request) const
+{
+	std::string path = filePath(request);
+	if (path.empty())
+		return false;
+	std::ifstream f(path.c_str());
+  return f.good();
+}
+
+bool FileHandler::checkIfResponsible(HttpRequest *request, const std::vector<std::string>& tokens) const
+{
+	return request->getMethod() == "GET"
+		&& doesFileExist(request);
+}
+
+static std::string getSuffix(std::string &path)
+{
+	auto i = path.rfind(".");
+	std::string suffix;
+	if (i != std::string::npos)
+		suffix = path.substr(i + 1);
+	return suffix;
+}
+
+static std::map<std::string, std::string> suffixMapping = {
+	{ "css", "text/css" },
+	{ "gif", "image/gif" },
+	{ "htm", "text/html" },
+	{ "html", "text/html" },
+	{ "jpg", "image/jpg" },
+	{ "js", "text/javascript" },
+	{ "pdf", "application/pdf" },
+	{ "png", "image/png" },
+	{ "txt", "text/plain" },
+	{ "xml", "application/xml" }
+};
+
+void FileHandler::handleRequest(HttpRequest *request, const std::vector<std::string>& tokens)
+{
+	std::string path = filePath(request);
+	std::ifstream f(path.c_str());
+	if (f.good())
+	{
+		std::string type = suffixMapping[getSuffix(path)];
```

Comment:
If suffix from path is not in suffixMapping it is created, with an empty string as value; is it really what you want?